core/aws_ddk_core/stages/appflow.py

```diff
@@ -0,0 +1,224 @@
+from aws_cdk.aws_events_targets import SfnStateMachine
+from aws_cdk.aws_iam import Effect, PolicyStatement, Role, ServicePrincipal
+from aws_cdk.aws_lambda import Code
+from aws_cdk.aws_stepfunctions import (
+    Choice,
+    Condition,
+    CustomState,
+    Fail,
+    StateMachine,
+    StateMachineType,
+    Succeed,
+    Wait,
+    WaitTime,
+)
+from aws_cdk.aws_stepfunctions_tasks import LambdaInvoke
+from aws_ddk_core.pipelines import DataStage
+from aws_ddk_core.resources import LambdaFactory, StepFunctionsFactory
+from constructs import Construct
+
+
+class AppFlowStage(DataStage):
+    """
+    Class that represents an AppFlow DDK DataStage.
+    """
+
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        environment_id: str,
+        flow_name: Optional[str] = None,
+        destination_flow_config: Optional[CfnFlow.DestinationFlowConfigProperty] = None,
+        source_flow_config: Optional[CfnFlow.SourceFlowConfigProperty] = None,
+        tasks: Optional[List[CfnFlow.TaskProperty]] = None,
+        state_machine_input: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        DDK AppFlow stage.
+
+        Stage that contains a step function that runs an AppFlow flow.
+        The flow must be pre-created.
+
+        Parameters
+        ----------
+        scope : Construct
+            Scope within which this construct is defined
+        id : str
+            Identifier of the stage
+        environment_id : str
+            Identifier of the environment
+        flow_name : Optional[str]
+            Name of the AppFlow flow to run
+        destination_flow_config : Optional[CfnFlow.DestinationFlowConfigProperty]
+            The flow destination_flow_config properties
+        source_flow_config : Optional[CfnFlow.SourceFlowConfigProperty]
+            The flow source_flow_config properties
+        tasks : Optional[List[CfnFlow.TaskProperty]]
+            The flow tasks properties
+        state_machine_input : Optional[Dict[str, Any]]
+            Input of the state machine
+        """
+        super().__init__(scope, id)
+
+        self._state_machine_input: Optional[Dict[str, Any]] = state_machine_input
+        self._event_detail_type: str = f"{id}-event-type"
+
+        # If AppFlow flow name is not supplied, create one
+        self._flow: Optional[CfnFlow] = None
+        if not flow_name:
+            self._flow = CfnFlow(
+                self,
+                f"{id}-flow",
+                destination_flow_config_list=[destination_flow_config],
+                source_flow_config=source_flow_config,
+                tasks=tasks,
+                trigger_config=CfnFlow.TriggerConfigProperty(trigger_type="OnDemand"),
+            )
+            flow_name = self._flow.name
+
+        # Create lambda used to check flow execution status
+        status_lambda_role = Role(
+            self,
+            "flow-execution-status-lambda-role",
```

Comment:
Could we add the `f"{id}` both here and in the lambda function?

Reply:
added